test/blackbox/api/dds-pim/PubSubWriter.hpp

```diff
@@ -197,27 +199,36 @@ class PubSubWriter
                 eprosima::fastdds::dds::DataWriter* datawriter,
                 const eprosima::fastrtps::OfferedDeadlineMissedStatus& status) override
         {
-            (void)datawriter;
+            static_cast<void>(datawriter);
             times_deadline_missed_ = status.total_count;
         }
 
         void on_offered_incompatible_qos(
                 eprosima::fastdds::dds::DataWriter* datawriter,
                 const eprosima::fastdds::dds::OfferedIncompatibleQosStatus& status) override
         {
-            (void)datawriter;
+            static_cast<void>(datawriter);
             writer_.incompatible_qos(status);
         }
 
         void on_liveliness_lost(
                 eprosima::fastdds::dds::DataWriter* datawriter,
                 const eprosima::fastrtps::LivelinessLostStatus& status) override
         {
-            (void)datawriter;
+            static_cast<void>(datawriter);
             times_liveliness_lost_ = status.total_count;
             writer_.liveliness_lost();
         }
 
+        void on_unacknowledged_sample_removed(
+                eprosima::fastdds::dds::DataWriter* datawriter,
+                const eprosima::fastdds::dds::InstanceHandle_t& handle) override
+        {
+            EXPECT_EQ(writer_.datawriter_, datawriter);
+            times_unack_sample_removed_++;
+            writer_.instances_removed_unack_.push_back(handle);
```

Comment:
We should be consistent and keep both data inside the listener. This means you would need to add a getter for `instances_removed_unack_`.